amazon/ionbenchmark/ion_benchmark_cli.py

```diff
@@ -84,52 +100,169 @@
+                    tracemalloc.stop()
+                    return data
+            else:
+                def test_func():
+                    tracemalloc.start()
+                    it = ion.loads(benchmark_data, single_value=single_value, emit_bare_values=emit_bare_values,
+                                   parse_eagerly=False)
                     while True:
                         try:
                             next(it)
                         except StopIteration:
                             break
-                return it
+                    global read_memory_usage_peak
+                    read_memory_usage_peak = tracemalloc.get_traced_memory()[1] / BYTES_TO_MB
+                    tracemalloc.stop()
+                    return it
     else:
-        if not iterator:
+        if not memory_profiling:
+            if not iterator:
+                def test_func():
+                    with open(file, "br") as fp:
+                        data = ion.load(fp, single_value=single_value, emit_bare_values=emit_bare_values)
+                    return data
+            else:
+                def test_func():
+                    with open(file, "br") as fp:
+                        it = ion.load(fp, single_value=single_value, emit_bare_values=emit_bare_values,
+                                      parse_eagerly=False)
+                        while True:
+                            try:
+                                next(it)
+                            except StopIteration:
+                                break
+                    return it
+        else:
+            if not iterator:
+                def test_func():
+                    tracemalloc.start()
+                    with open(file, "br") as fp:
+                        data = ion.load(fp, single_value=single_value, emit_bare_values=emit_bare_values)
+                    global read_memory_usage_peak
+                    read_memory_usage_peak = tracemalloc.get_traced_memory()[1] / BYTES_TO_MB
+                    tracemalloc.stop()
+                    return data
+            else:
+                def test_func():
+                    tracemalloc.start()
+                    with open(file, "br") as fp:
+                        it = ion.load(fp, single_value=single_value, emit_bare_values=emit_bare_values,
+                                      parse_eagerly=False)
+                        while True:
+                            try:
+                                next(it)
+                            except StopIteration:
+                                break
+                    global read_memory_usage_peak
+                    read_memory_usage_peak = tracemalloc.get_traced_memory()[1] / BYTES_TO_MB
+                    tracemalloc.stop()
+                    return it
+    return test_func
+
+
+# Generates benchmark code for json/cbor load/loads APIs
+def generate_read_test_code(file, memory_profiling, format_option, binary, io_type):
```

Comment:
Is there a reason we can't follow the same path for Ion?

Reply:
Merged them into one.